lib/src/gateway/gateway.dart

```diff
@@ -144,6 +144,9 @@ class Gateway extends GatewayManager with EventParser {
             // ignore: avoid_function_literals_in_foreach_calls
             messageIds.forEach((messageId) => MessageManager(client.options.messageCacheConfig, client, channelId: channelId).cache.remove(messageId)),
           UserUpdateEvent(:final user) => client.users.cache[user.id] = user,
+          StageInstanceCreateEvent(:final instance) || StageInstanceUpdateEvent(:final instance) => client.channels.stageInstanceCache[instance.channelId] =
+              instance,
+          StageInstanceDeleteEvent(:final instance) => client.channels.stageInstanceCache.remove(instance.channelId),
```

Comment:
`:final instance` what is that syntax? Is it something new?

Reply:
It's new in Dart 3.0 - this entire code block is the new switch expressions, and each case is a pattern, another new feature (there are new pages on dart.dev for patterns if you want to read more).

In this case, this pattern indicates to match any instance of StageCreateEvent and create a local variable called `instance` which contains the value of the event's `instance` field.

Reply:
nice, thanks